### PR TITLE
Bug #210287 - [WEB]: Prerak Attendance -> Multiple entries are getting added in the database when marking the attendance as present and absent for same date

### PR DESCRIPTION
**Pull request by sagarkoshti1990**
Bug #210287 -: https://tracker.tekdi.net/issues/210287

### I have ensured that following `Pull Request Checklist` is taken care of before sending this PR
* [ ] Code is formatted as per format decided
* [ ] Updated acceptance criteria in tracker
* [ ] Updated test cases in test-cases-tracker

## Summary by CodeRabbit

- **New Features**
  - Enhanced attendance tracking with updated logic.
  - Location-based UI modifications for a more personalized experience.
  - Improved facilitator details presentation.

- **Improvements**
  - Refined user input handling for better interaction.
  - Updated modal behavior for increased usability.

- **Bug Fixes**
  - Adjusted certificate generation and display process for accuracy and reliability.

### CODE REVIEW

File: apps/front-end/src/pages/front-end/Attendence/Attendence.js [hunk 1]
```diff
@@ -201,57 +190,52 @@ const RenderAttendanceColumn = React.memo(({ row }) => {
...
+          : attendance?.status === "absent"
           ? "Absent"
           : "Mark"}
-      </Text>
+      </AdminTypo.H7>
       <Switch
+        key={attendance}
         isDisabled={isDisabledAttBtn === `${row.id}-${row.presentDate}`}
         offTrackColor="dangerColor"
         onTrackColor="successColor"
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply apps/front-end/src/pages/front-end/Attendence/Attendence.js:241]: chk this @sagarkoshti1990

File: apps/front-end/src/pages/front-end/Attendence/Attendence.js [hunk 2]
```diff
@@ -97,41 +96,32 @@ const renderStatusColumn = (row, t) => (
...
+        date_time: row?.presentDate,
+        status: row?.attendance_status,
+      };
+      return await attendanceService.createAttendance(data);
     }
-    rounded="sm"
-  />
-);
+  }
+};
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply apps/front-end/src/pages/front-end/Attendence/Attendence.js:124]: chk this @sagarkoshti1990

File: lib/common-lib/src/services/EventService.js [hunk 3]
```diff
@@ -95,7 +95,7 @@ export const updateAttendance = async (data = {}, header = {}) => {
       }
     )
     if (result?.data) {
-      return result
+      return result?.data
     } else {
       return {}
     }
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply lib/common-lib/src/services/EventService.js:101]: @sagarkoshti1990 chk this

There appears to be a typo in the import section: distory should be corrected to destroy if that is the intended function from the RestClient. This typo could lead to runtime errors if not addressed.

import { distory } from './RestClient'
=>
import { destroy } from './RestClient'

File: apps/front-end/src/pages/front-end/Attendence/Attendence.js [hunk 4]
```diff
@@ -373,12 +359,12 @@ export default function Attendence({ footerLinks }) {
       console.log("Geolocation is not supported by this browser.");
```
Comment (swapnilphalke03) [inline apps/front-end/src/pages/front-end/Attendence/Attendence.js:359]: Can we remove console.log.

**Review by swapnilphalke03 — CHANGES_REQUESTED** [review]

**Review by swapnilphalke03 — APPROVED** [review]

**Pull request merged**